wire-runtime/src/main/java/com/squareup/wire/Message.java

```diff
@@ -160,46 +161,40 @@ protected void setBuilder(Builder builder) {
     return adapter.fromInt(value);
   }
 
+  /**
+   * Create a {@code byte[]} of the serialized form of the data.
+   * <p>
+   * Note: This is an inefficient means of serializing data. For best performance, call
+   * {@link Wire#adapter} and then {@link MessageAdapter2#write}.
+   */
   @SuppressWarnings("unchecked")
   public byte[] toByteArray() {
```

Comment:
I would prefer to leave it. The analogy with JSON is flawed, the only common supertype for JSON objects is `Object`. This provides a simple way in to the API. Once the developer gets Wire working in their app, they can learn about MessageAdapter, Okio streams, etc., to make things more efficient.